doc/src/build/prog-trans-ts-sdk.md

```diff
@@ -0,0 +1,234 @@
+---
+title: Sui Programmable Transaction with the TS SDK
+---
+
+In Sui, all user-initiated transactions are called Programmable Transactions.
+
+## Get Started
+
+First, make sure that you have the latest TypeScript SDK installed.
+
+We’ll start by constructing a transaction to send Sui. If you are familiar with the legacy transaction types, this is similar to a `paySui` transaction. You can start constructing transactions by importing the `Transaction` class, and constructing it:
+
+```tsx
+import { Transaction } from '@mysten/sui.js';
+const tx = new Transaction();
+```
+
+Using this, you can then add commands to this transaction.
+
+```tsx
+// Create a new coin with balance 100, based on the coins used as gas payment.
+// You can define any balance here.
+const coin = tx.splitCoin(tx.gas, tx.pure(100));
+
+// Transfer the split coin to a specific address.
+tx.transferObjects([coin], tx.pure('0xSomeSuiAddress'));
+```
+
+Note that you can attach multiple of the same command type to a transaction as well. For example, we can get a list of transfers, and iterate over them to transfer coins to each of them:
+
+```tsx
+interface Transfer {
+  to: string;
+  amount: number;
+}
+
+// Procure a list of some Sui transfers we need to make:
+const transfers: Transfer[] = getTransfers();
+
+const tx = new Transaction();
+
+// Create commands for each transfer
+transfers.forEach((transfer) => {
+  const coin = tx.splitCoin(tx.gas, tx.pure(transfer.amount));
+  tx.transferObjects([coin], tx.pure(transfer.to));
+});
+```
+
+After you have the transaction defined, you can directly execute it with a signer using `signAndExecuteTransaction`.
+
+```tsx
+signer.signAndExecuteTransaction({ transaction: tx });
+```
+
+## Inputs and commands
+
+Programmable Transactions have two key concepts: inputs and commands.
+
+Inputs are values that are used as arguments to the commands in the transaction. Inputs can either be an object reference (either to an owned object, or a shared object), or a pure BCS value (for example, an encoded string used as an argument to a move call).
+
+Commands are steps of execution in the transaction. You can also use the result of previous commands as arguments to a future command. By combining multiple commands together, programmable transactions provide a flexible way to create complex transactions.
```

Comment:
I really love this phrasing! I always struggle to explain things like this. I want to print this out and hang it on my wall as a reminder, lol